Source/Adobe.Target.Client/Service/OnDeviceDecisioningService.cs

```diff
@@ -143,10 +154,19 @@ internal OnDeviceDecisioningEvaluation EvaluateLocalExecution(TargetDeliveryRequ
         {
             foreach (var details in detailsList)
             {
-                var detailsContext =
-                    this.CollateParams(DetailsParamsCollators, deliveryRequest, details.GetRequestDetails());
+                var detailsContext = new Dictionary<string, object>(requestContext);
+                detailsContext.AddAll(this.CollateParams(DetailsParamsCollators, deliveryRequest, details.GetRequestDetails()));
 
-                // TODO: call decision handler
+                this.decisionHandler.ExecuteDetails(
```

Comment:
Something is not right if we have to pass so many parameters to a single method.

Reply:
Ok, I'll see if I can extract a DTO..

Reply:
As discussed, we'll address this in a future PR